lib/URI/_generic.pm

```diff
@@ -10,11 +10,41 @@ use Carp ();
 
 our $VERSION = '5.11';
 
-my $ACHAR = $URI::uric;  $ACHAR =~ s,\\[/?],,g;
-my $PCHAR = $URI::uric;  $PCHAR =~ s,\\[?],,g;
+my $ACHAR = URI::HAS_RESERVED_SQUARE_BRACKETS ? $URI::uric : $URI::uric4host;  $ACHAR =~ s,\\[/?],,g;
+my $PCHAR = $URI::uric;                                                        $PCHAR =~ s,\\[?],,g;
 
 sub _no_scheme_ok { 1 }
 
+our $IPv6_re;
+{ #-- "borrowed" from SALVAs Regexp::IPv6 - https://metacpan.org/dist/Regexp-IPv6/source/lib/Regexp/IPv6.pm
+  #TODO: Should be made a dependency of this module.
```

Comment:
I think that would require more discussion.

Reply:
Agreed, that's why I pasted the checks into the file instead of introducing another library.
See also my first reply (#TODO:) .